src/contract/contract_function_parameters.rs

```diff
@@ -190,10 +208,10 @@ impl ContractFunctionParameters {
     }
 
     /// Add a `bytes32` argument to the `ContractFunctionParameters`
-    pub fn add_bytes32(&mut self, val: &[u8; 32]) -> &mut Self {
+    pub fn add_bytes32<T: AsBytes32>(&mut self, val: &T) -> &mut Self {
```

Comment:
if you're getting errors about `Sized` when it comes to passing a `&str`, you can make it `T: AsBytes32 + ?Sized`

Reply:
@izik1 Oh, I didn't add `impl Sized for str` in the `mod private {}` at first. This was giving me a problems yesterday 😅. I made the changes just now!